sdcm/cluster_aws.py

```diff
@@ -643,6 +644,22 @@ def restart(self):
             return
 
         event_filters = ()
+        if self.is_seed:
+            # Due to https://github.com/scylladb/scylla/issues/7588, when we restart a node that is defined as "seed",
+            # we must state a different, living node as the seed provider in the scylla yaml of the restarted node
+            other_nodes = list(set(self.parent_cluster.nodes) - {self})
+            free_nodes = [node for node in other_nodes if not node.running_nemesis]
+            random_node = random.choice(free_nodes)
+
+            seed_provider = [{
+                "class_name": "org.apache.cassandra.locator.SimpleSeedProvider",
+                "parameters": [{
+                    "seeds": f"{random_node.ip_address}"
+                }, ],
+            }, ]
+
+            with self.remote_scylla_yaml() as scylla_yml:
+                scylla_yml["seed_provider"] = seed_provider
```

Comment:
Need the same for other clusters types, not only aws.

Reply:
Right now other backend doesn't support this nemesis, i.e. the replace flow is only supported by AWS backend
Moving this only the seed logic into shared place, won't be helpful at this stage.